igor-web/src/test/groovy/com/netflix/spinnaker/igor/build/InfoControllerSpec.groovy

```diff
@@ -142,6 +142,7 @@ class InfoControllerSpec extends Specification {
 
         then:
         1 * buildMasters.filteredMap(BuildServiceProvider.JENKINS) >> ["travis-master1": []]
+        1 * buildMasters.filteredMap(BuildServiceProvider.WERCKER) >> ["travis-master1": []]
```

Comment:
I'm not sure I understand this test case---why are we adding "travis-master1" to the Wercker masters in the case that's testing Travis?  I think this test case should be exercising the Travis code, and a new test case should exercise the Wercker code.

(Also, this is pre-existing, but I'm not sure why the above line adds the master to Jenkins, not Travis.)

Reply:
I saw an issue in InfoController that the line of buildMasters.filteredMap(BuildServiceProvider.JENKINS)[master] could cause NPE when
buildMasters.filteredMap(BuildServiceProvider.JENKINS) returns null, instead of empty map, which I guess is for the case that there is no jenkins master. I am not sure if this null would happen only in the Mock(BuildMasters) or also in real BuildMasters. Those two lines of adding Jenkins and Wercker were to avoid the NPE. I changed 
InfoController to avoid NPE in case the real BuildMasters.filteredMap does return null.

I reverted this test to only adding the Jenkins though that line of adding jenkins is no longer needed (to avoid NPE). I also added a test for wercker.

Reply:
Ok, that makes sense---both the fix for the NPE in `InfoController.groovy` and the updates to the tests; thanks for the explanation.